packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1582,6 +1582,26 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   static RenderObject? get debugActiveLayout => _debugActiveLayout;
   static RenderObject? _debugActiveLayout;
 
+  /// Set [debugActiveLayout] to null when [inner] callback is called.
+  /// This is useful when you have to temporarily clear that variable to
+  /// disable some false-positive checks, such as when computing toStringDeep
+  /// or using custom trees
+  @pragma('vm:prefer-inline')
+  static T _debugWithActiveLayoutCleared<T>(T Function() inner) {
```

Comment:
maybe name this `_withDebugActiveLayoutCleared`? Methods starting with `debug` or `_debug` should only be called from within an `assert` or guarded by kIsDebug. This doesn't have that restriction, though.

Reply:
Agree, renamed (will push code in a minute)